### ISSUE TITLE
ignorePattern for max-line-length not working as expected

### ISSUE THREAD

**Issue by 1BE6-ankit** [stylelint/stylelint#4519]
> Clearly describe the bug

I have a max-line-length rule with ignorePattern in my config file.

This rule is supposed to ignore the max-line-length for lines that have the comment of form: `/*.... */`. But when I run stylelint with the stylelint command and pass the .stylelintrc file, an error is thrown.

> Which rule, if any, is the bug related to?

max-line-length (with ignorePattern)

> What CSS is needed to reproduce the bug?

No CSS is required.

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    	"max-line-length": [120, {
		ignorePattern: ["/\/\*.*?\*\//"]
	}],
  }
}
```

> Which version of stylelint are you using?

`12.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"CLI with `stylelint "**/*.css" --config myconfig.json`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

Stylelint was supposed to lint the input files, but it throws error beforehand.

> What actually happened (e.g. what warnings or errors did you get)?

Stylelint could not parse the regex included in ignorePattern.
I have tested the regex pattern with online tools like: https://www.regextester.com/, https://regex101.com/ and others. It works fine with them. 

```shell
YAMLSyntaxError: YAML Error in C:\Users\koirala\Desktop\projects\IDE\linting\SCSS\stylelint\.stylelintrc:
Invalid escape sequence \* at line 17, column 20:

                        ignorePattern: ["/\/\*.*?\*\//"]
                   ^^^^^^^^^^^^^^^
```

**Comment by hudochenkov** [comment]
You have invalid JSON (it's strange that you get YAML error). Even if fix obvious syntax mistakes it still not valid with pattern that you added:

```json
{
  "rules": {
    	"max-line-length": [120, {
		"ignorePattern": ["/\/\*.*?\*\//"]
	}]
  }
}
```

https://jsonlint.com/

You might need to escape escape characters (do `\\` instead of `\`):

```json
{
	"rules": {
		"max-line-length": [120, {
			"ignorePattern": ["/\\/\\*.*?\\*\\//"]
		}]
	}
}
```

This is valid JSON, but I'm not sure if it will work as you want.

**Comment by 1BE6-ankit** [comment]
It works now. Thanks.
